dev/src/components/Groups.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from "react"
+import { Group } from "../../proto/ag/ag_pb"
+import { getCourseID, hasEnrollments } from "../Helpers"
+import { useActions, useAppState } from "../overmind"
+import GroupForm from "./group/GroupForm"
+import Search from "./Search"
+
+/* Lists all groups for a given course. */
+export const Groups = (): JSX.Element => {
+    const state = useAppState()
+    const actions = useActions()
+    const courseID = getCourseID()
+
+
+    const [editing, setEditing] = useState<Group>()
+
+    const groupSearch = (group: Group) => {
+        // Show all groups if query is empty
+        if (state.query.length == 0) {
+            return false
+        }
+
+        // Show group if group name includes query
+        if (group.getName().toLowerCase().includes(state.query)) {
+            return false
+        }
+
+        // Show group if any group user includes query
+        for (const user of group.getUsersList()) {
+            if (user.getName().toLowerCase().includes(state.query)) {
+                return false
+            }
+        }
+        // Hide group if none of the above include query
+        return true
+    }
+
+    const updateGroupStatus = (group: Group, status: Group.GroupStatus) => {
+        actions.updateGroupStatus({ group, status })
+    }
+
+    const GroupButtons = ({ group }: { group: Group }) => {
+        if (group.getStatus() === Group.GroupStatus.PENDING) {
+            return (
+                <td>
+                    <span onClick={() => updateGroupStatus(group, Group.GroupStatus.APPROVED)} className="badge badge-primary clickable">Approve</span>
+                    <span className="badge badge-info clickable ml-2" onClick={() => setEditing(group)}>Edit</span>
+                    <span onClick={() => actions.deleteGroup(group)} className="badge badge-danger clickable ml-2">Delete</span>
+                </td>
+            )
+        }
+        return <td><span className="badge badge-info clickable" onClick={() => setEditing(group)}>Edit</span></td>
+    }
+
+    const GroupRow = ({ group }: { group: Group }) => {
+        return (
+            <>
+                <tr hidden={groupSearch(group)}>
+                    <th key={group.getId()}>
+                        {group.getName()}
+                        <span className="badge badge-warning ml-2">{group.getStatus() == Group.GroupStatus.PENDING ? "Pending" : null}</span>
```

Comment:
Could add a helper for `isPendingGroup: group.getStatus() === Group.GroupStatus.PENDING`